components/automate-cli/pkg/diagnostics/integration/iam_projects.go

```diff
@@ -130,16 +130,19 @@ func GetProject(tstCtx diagnostics.TestContext, id string) (*ProjectInfo, error)
 	return &projectInfo, nil
 }
 
-// DeleteProject deletes the project with the given id
-func DeleteProject(tstCtx diagnostics.TestContext, id string) error {
-	// any associated rules are cascade-deleted with the project
-	err := MustJSONDecodeSuccess(
-		tstCtx.DoLBRequest(
-			fmt.Sprintf("/apis/iam/v2/projects/%s", id),
-			lbrequest.WithMethod("DELETE"),
-		)).Error()
+// DeleteProjectWithRule deletes the project with the given id and its associated rule
+func DeleteProjectWithRule(tstCtx diagnostics.TestContext, projectID, ruleID string) error {
+	if err := MustJSONDecodeSuccess(tstCtx.DoLBRequest(
+		fmt.Sprintf("/apis/iam/v2/projects/%s/rules/%s", projectID, ruleID),
+		lbrequest.WithMethod("DELETE"),
+	)).WithValue(&struct{}{}); err != nil {
```

Comment:
`.Error()` was only checking the err of `resp, err := ...`. Using `WithValue(&struct{}{})`, we ignore whatever's returned (the struct we marshal into has no fields), but we leverage that the `WithValue` method does check the HTTP status return for code 200.

Reply:
good catch 👍